k9mail/src/main/java/com/fsck/k9/activity/setup/autoconfiguration/AccountSetupAutoConfigurationPresenter.java

```diff
@@ -0,0 +1,576 @@
+                                Timber.w("unsupported SubjectAltName of type x400Address");
+                                continue;
+                            case 4:
+                                Timber.w("unsupported SubjectAltName of type directoryName");
+                                continue;
+                            case 5:
+                                Timber.w("unsupported SubjectAltName of type ediPartyName");
+                                continue;
+                            case 6:  // Uri
+                                name = (String)value;
+                                break;
+                            case 7: // ip-address
+                                name = (String)value;
+                                break;
+                            default:
+                                Timber.w("unsupported SubjectAltName of unknown type");
+                                continue;
+                        }
+
+                        // if some of the SubjectAltNames match the store or transport -host,
+                        // display them
+                        if (name.equalsIgnoreCase(storeURIHost) || name.equalsIgnoreCase(transportURIHost)) {
+                            //TODO: localize this string
+                            altNamesText.append("Subject(alt): ").append(name).append(",...\n");
+                        } else if (name.startsWith("*.") && (
+                                storeURIHost.endsWith(name.substring(2)) ||
+                                        transportURIHost.endsWith(name.substring(2)))) {
+                            //TODO: localize this string
+                            altNamesText.append("Subject(alt): ").append(name).append(",...\n");
+                        }
+                    }
+                    chainInfo.append(altNamesText);
+                }
+            } catch (Exception e1) {
+                // don't fail just because of subjectAltNames
+                Timber.w(e1, "cannot display SubjectAltNames in dialog");
+            }
+
+            chainInfo.append("Issuer: ").append(chain[i].getIssuerDN().toString()).append("\n");
+            if (sha1 != null) {
+                sha1.reset();
+                try {
+                    String sha1sum = Hex.encodeHex(sha1.digest(chain[i].getEncoded()));
+                    chainInfo.append("Fingerprint (SHA-1): ").append(sha1sum).append("\n");
+                } catch (CertificateEncodingException e) {
+                    Timber.e(e, "Error while encoding certificate");
+                }
+            }
+
+            view.showAcceptKeyDialog(msgResId, exMessage, chainInfo.toString(), chain[0]);
+        }
+
+    }
+
+    private void handleCertificateValidationException(CertificateValidationException cve) {
+        Timber.e(cve, "Error while testing settings");
+
+        X509Certificate[] chain = cve.getCertChain();
+        // Avoid NullPointerException in acceptKeyDialog()
```

Comment:
This comment is silly - obviously checking for null avoids an NPE. Doesn't really tell us why it would be null.